### PR TITLE
Fix bad documentation of quantity errors in event objects

### PR DESCRIPTION
**Pull request by megies**
Also improved class structure for QuantityError, and warning on setting a non-expected attribute (I think we should do this for all Event-type objects -- quite a few times I've seen people wondering why they're setting some attribute wasn't working, just to find out that it was just a typo in the attribute name).

### CODE REVIEW

File: obspy/core/event/base.py [hunk 1]
```diff
@@ -40,10 +40,41 @@
...
+    :param lower_uncertainty: Uncertainty as the absolute value of deviation
+        from the main value towards smaller values.
+    :type upper_uncertainty: float
+    :param upper_uncertainty: Uncertainty as the absolute value of deviation
+        from the main value towards larger values.
+    :type confidence_level: float
+    :param confidence_level: Confidence level of the uncertainty, given in
+        percent (0-100).
+    """
+    defaults = {"uncertainty": None, "lower_uncertainty": None,
```
Comment (QuLogic) [inline obspy/core/event/base.py]: I'm not sure I understand why this is a dictionary?

File: obspy/core/event/base.py [hunk 2]
```diff
@@ -40,10 +40,41 @@
...
+    def __init__(self, uncertainty=None, lower_uncertainty=None,
+                 upper_uncertainty=None, confidence_level=None):
+        super(QuantityError, self).__init__()
+        self.uncertainty = uncertainty
+        self.lower_uncertainty = lower_uncertainty
+        self.upper_uncertainty = upper_uncertainty
+        self.confidence_level = confidence_level
+
+    def __bool__(self):
+        if any([getattr(self, key) is not None for key in self.defaults]):
```
Comment (QuLogic) [inline obspy/core/event/base.py]: Can you not just return this? Why the `if`?

File: obspy/core/event/base.py (hunk 1 above)
Comment (megies) [inline obspy/core/event/base.py]: I don't know either.. in any case that's the current implementation of AttribDict..

File: obspy/core/event/base.py (hunk 2 above)
Comment (megies) [inline obspy/core/event/base.py]: Yeah.. right. I'll change it..

File: obspy/core/event/base.py (hunk 1 above)
Comment (megies) [inline obspy/core/event/base.py]: Ah.. that's for the `Stats` default values for `sampling_rate` etc. I think..

**Comment by krischer** [comment]
Maybe add a quick test?

**Comment by megies** [comment]
For the new `__bool__()`? 5f06227 OK?

**Comment by krischer** [comment]
I originally meant a test for the new raised warning but testing `__bool__()` is of course also good to have :-)

**Comment by megies** [comment]
OK done. I think we should issue a warning for all Event-type objects (i.e. set `defaults` and `warn_on_non_default_key=True` in the event type factory).

**Pull request merged**